gusto/wrappers.py

```diff
@@ -33,6 +33,9 @@ def __init__(self, time_discretisation, wrapper_options):
         self.time_discretisation = time_discretisation
         self.options = wrapper_options
         self.solver_parameters = None
+        self.idx = None
+        self.mixed_options = False
```

Comment:
I'm a little bit uncertain about having a general `Wrapper` object know about `mixed_options`...

It feels like the inheritance of properties could be the wrong way around, as a `MixedFSWrapper` has several subwrappers, but the subwrappers then need to know that they are subwrapper. Ideally it would be better if they don't know that they are subwrappers!

Looking below at why you've had to do this, it's to set the function space. I'm just wondering if we can find a cleaner way to set `original_space`? Maybe `idx` could be an argument to the `__init__` or `setup` method of the base `Wrapper` class?